src/equations/shallow_water_two_layer_2d.jl

```diff
@@ -273,274 +273,82 @@ end
-
-  # Create average and jump values
-  h_upper_average = 0.5 * (h_upper_ll + h_upper_rr)
-  h_lower_average = 0.5 * (h_lower_ll + h_lower_rr)
-  h_upper_jump    = h_upper_rr - h_upper_ll
-  h_lower_jump    = h_lower_rr - h_lower_ll
-  b_jump     = b_rr  - b_ll
-
-  # Assign variables for constants for better readability
-  g = equations.gravity
-
-  # Bottom gradient nonconservative term: (0, g*h_upper*(b+h_lower)_x, g*h_upper*(b+h_lower)_y, 0,
-  #                                        g*h_lower*(b+r*h_upper)_x, g*h_lower*(b+r*h_upper)_x, 0)
-
-  # Includes two parts:
-  #   (i)  Diagonal (consistent) term from the volume flux that uses `b_ll` to avoid
-  #        cross-averaging across a discontinuous bottom topography
-  #   (ii) True surface part that uses `h_average` and `b_jump` to handle discontinuous bathymetry
-  z = zero(eltype(u_ll))
-  if orientation == 1 
-    f = SVector(
-      z,
-      g * h_upper_ll * (b_ll +   h_lower_ll) + g * h_upper_average * (b_jump +   h_lower_jump),
-      z,z,
-      g * h_lower_ll * (b_ll + equations.r * h_upper_ll) + g * h_lower_average * (b_jump +
-          equations.r * h_upper_jump),
-      z,z)
-  else # orientation == 2
-    f = SVector(
-      z,z,
-      g * h_upper_ll * (b_ll +   h_lower_ll) + g * h_upper_average * (b_jump +   h_lower_jump),
-      z,z,
-      g * h_lower_ll * (b_ll + equations.r * h_upper_ll) + g * h_lower_average * (b_jump + 
-          equations.r * h_upper_jump),
-      z)
-  end
+# Pull the necessary left and right state information
+h_upper_ll, h_lower_ll = waterheight(u_ll, equations)
+h_upper_rr, h_lower_rr = waterheight(u_rr, equations)
+b_rr = u_rr[7]
+b_ll = u_ll[7]
+
+# Calculate jumps
+h_upper_jump =  (h_upper_rr - h_upper_ll)
+h_lower_jump =  (h_lower_rr - h_lower_ll)
+b_jump       =  (b_rr       - b_ll)
+
+z = zero(eltype(u_ll))
+
+# Bottom gradient nonconservative term: (0, g*h_upper*(b + h_lower)_x, g*h_upper*(b + h_lower)_y ,
+#                                        0, g*h_lower*(b + r*h_upper)_x, 
+#                                        g*h_lower*(b + r*h_upper)_y, 0)
+if orientation == 1
+  f = SVector(z,
+  equations.gravity * h_upper_ll * (b_jump + h_lower_jump),
+  z,z,
+  equations.gravity * h_lower_ll * (b_jump + equations.r * h_upper_jump),
+  z,z)
+else # orientation == 2
+  f = SVector(z, z,
+  equations.gravity * h_upper_ll * (b_jump + h_lower_jump),
+  z,z,
+  equations.gravity * h_lower_ll * (b_jump + equations.r * h_upper_jump),
+  z)
+end
 
-  return f
+return f
 end
 
-@inline function flux_nonconservative_fjordholm_etal(u_ll, u_rr,
+@inline function flux_nonconservative_ersing_etal(u_ll, u_rr,
                                                      normal_direction_ll::AbstractVector,
                                                      normal_direction_average::AbstractVector,
                                                      equations::ShallowWaterTwoLayerEquations2D)
```

Comment:
```suggestion
@inline function flux_nonconservative_ersing_etal(u_ll, u_rr,
                                                  normal_direction_ll::AbstractVector,
                                                  normal_direction_average::AbstractVector,
                                                  equations::ShallowWaterTwoLayerEquations2D)
```

Reply:
Done in 7d75332d1eb1396dc1d0070ec2f53c8d2501853c